client/my-sites/email/email-management/home/email-plan-mailboxes-list.jsx

```diff
@@ -109,7 +131,122 @@ const getActionsForMailbox = ( mailbox, translate, dispatch ) => {
 	};
 };
 
-function EmailPlanMailboxesList( { mailboxes, isLoadingEmails } ) {
+const TitanActionMenu = () => {
+	const translate = useTranslate();
+
+	return (
+		<EllipsisMenu
+			popoverClassName="email-plan-mailboxes-list__mailbox-action-menu-popover"
+			className="email-plan-mailboxes-list__mailbox-action-menu"
+		>
+			<PopoverMenuItem
+				href="https://wp.titan.email/mail/"
+				className="email-plan-mailboxes-list__mailbox-action-menu-item"
+				isExternalLink={ true }
+			>
+				<img src={ titanMailIcon } alt={ translate( 'Titan Mail icon' ) } />
+				{ translate( 'Mail', {
+					comment: 'This refers to the Email application (i.e. the webmail) of Titan',
+				} ) }
+			</PopoverMenuItem>
+			<PopoverMenuItem
+				href="https://wp.titan.email/calendar/"
+				className="email-plan-mailboxes-list__mailbox-action-menu-item"
+				isExternalLink={ true }
+			>
+				<img src={ titanCalendarIcon } alt={ translate( 'Titan Calendar icon' ) } />
+				{ translate( 'Calendar', {
+					comment: 'This refers to the Calendar application of Titan',
+				} ) }
+			</PopoverMenuItem>
+			<PopoverMenuItem
+				href="https://wp.titan.email/contacts/"
+				className="email-plan-mailboxes-list__mailbox-action-menu-item"
+				isExternalLink={ true }
+			>
+				<img src={ titanContactsIcon } alt={ translate( 'Titan Contacts icon' ) } />
+				{ translate( 'Contacts', {
+					comment: 'This refers to the Contacts application of Titan',
+				} ) }
+			</PopoverMenuItem>
+		</EllipsisMenu>
+	);
+};
+const GmailActionMenu = ( { domainName } ) => {
+	const translate = useTranslate();
+
+	return (
+		<EllipsisMenu
+			className="email-plan-mailboxes-list__mailbox-action-menu"
+			popoverClassName="email-plan-mailboxes-list__mailbox-action-menu-popover"
+		>
+			<PopoverMenuItem
+				className="email-plan-mailboxes-list__mailbox-action-menu-item"
+				isExternalLink={ true }
+				href={ getGmailUrl( domainName ) }
```

Comment:
Another thought: I think we should add some of these links (like Gmail) using the full email address, not just the domain name.
I am not sure if Titan also supports that.

Reply:
By including the full email address, do you mean we need to pass Gmail instead of domainName to the `getGmailUrl` function? If you're talking about email address auto-fill, that seemed to be happening irrespective of whether we pass down `email` or `domainName` in the `getGmailUrl` method.

Pretty sure Titan doesn't support autofill through the generated URL as of now, but I can ask someone to do that for us.

Reply:
Not urgent if Titan doesn't support that yet.

But I think the complication will be when there are multiple email addresses -- specifying the email address is likely to be the right option if we're showing this menu alongside each mailbox name.

Reply:
And yes, I did mean passing `email` to the `getGmailUrl()` function for _all_ the Google links.

Reply:
Done, now passing `email` instead of `domainName` to the GSuite utilities.